server/apple_notification_server.go

```diff
@@ -38,6 +42,32 @@ func (me *AppleNotificationServer) Initialize() bool {
 			me.AppleClient = apns.NewClient(appleCert).Production()
 		}
 
+		// Override the native transport.
+		if CfgPP.ProxyServer != "" {
+			tlsConfig := &tls.Config{
+				Certificates: []tls.Certificate{appleCert},
+			}
+
+			if len(appleCert.Certificate) > 0 {
+				tlsConfig.BuildNameToCertificate()
+			}
+
+			transport := &http.Transport{
+				TLSClientConfig: tlsConfig,
+				Proxy: func(request *http.Request) (*url.URL, error) {
+					return url.Parse(CfgPP.ProxyServer)
+				},
+				IdleConnTimeout: apns.HTTPClientTimeout,
+			}
+			err := http2.ConfigureTransport(transport)
```

Comment:
Am I understanding correctly that Apple requires HTTP 2 for sending the push notifications to their servers?

Out of band from this PR, but we may need to check if $BIG_ENTERPRISE_CUSTOMER's HTTP proxy is compatible with HTTP 2.

Reply:
Correct.

> Out of band from this PR, but we may need to check if $BIG_ENTERPRISE_CUSTOMER's HTTP proxy is compatible with HTTP 2.

Yes, I have talked with Stu, and the $BIG_CUSTOMER will move to a new proxy which supports HTTP/2. They are aware of this.